mscore/zoombox.cpp

```diff
@@ -47,37 +47,35 @@ const std::array<ZoomEntry, 13> zoomEntries { {
      {  ZoomIndex::ZOOM_FREE,          0, "" },
      } };
 
-static constexpr ZoomIndex startZoomIndex = ZoomIndex::ZOOM_PAGE_WIDTH;
-
 ZoomState ZoomBox::getDefaultLogicalZoom()
       {
-      ZoomIndex index = startZoomIndex;
-      qreal logicalLevel = 1.0;
+      ZoomState result { ZoomIndex::ZOOM_100, 1.0 };
 
       // Convert the default-zoom preferences into a usable zoom index and logical zoom level.
       switch (static_cast<ZoomType>(preferences.getInt(PREF_UI_CANVAS_ZOOM_DEFAULT_TYPE))) {
-            case ZoomType::WHOLE_PAGE:
-                  index = ZoomIndex::ZOOM_WHOLE_PAGE;
-                  break;
-            case ZoomType::TWO_PAGES:
-                  index = ZoomIndex::ZOOM_TWO_PAGES;
-                  break;
             case ZoomType::PERCENTAGE: {
                   // Select a numeric preset zoom entry if the percentage corresponds to one; otherwise, select free zoom.
                   const auto logicalLevelPercentage = preferences.getInt(PREF_UI_CANVAS_ZOOM_DEFAULT_LEVEL);
                   const auto i = std::find(zoomEntries.cbegin(), zoomEntries.cend(), logicalLevelPercentage);
-                  index = ((i != zoomEntries.cend()) && i->isNumericPreset()) ? i->index : ZoomIndex::ZOOM_FREE;
-                  logicalLevel = logicalLevelPercentage / 100.0;
+                  result.index = ((i != zoomEntries.cend()) && i->isNumericPreset()) ? i->index : ZoomIndex::ZOOM_FREE;
+                  result.level = logicalLevelPercentage / 100.0;
                   }
                   break;
             case ZoomType::PAGE_WIDTH:
-                  Q_FALLTHROUGH();
+                  result.index = ZoomIndex::ZOOM_PAGE_WIDTH;
+                  break;
+            case ZoomType::WHOLE_PAGE:
+                  result.index = ZoomIndex::ZOOM_WHOLE_PAGE;
+                  break;
+            case ZoomType::TWO_PAGES:
+                  result.index = ZoomIndex::ZOOM_TWO_PAGES;
+                  break;
             default:
-                  index = ZoomIndex::ZOOM_PAGE_WIDTH;
+                  Q_ASSERT(false);
```

Comment:
This will crash in debug builds and there only, as I guess you to be aware of. Dropping the entire default part though would emit a compiler if a new value got added to the enum but not handled here.

Reply:
The intention is indeed to crash in debug builds (and only in debug builds) if an `enum` value is ever added but not explicitly handled.

I prefer this to leaving out the `default` case and relying on a warning as it more explicitly conveys the intention. What do you think?